### ISSUE TITLE
Get-BCArtifactUrl -select Daily on the day of a new version shipped

### ISSUE THREAD

**Issue by Arthurvdv** [microsoft/navcontainerhelper#3047]
**Describe the issue**
I think the behavior of the '-select Daily' parameter has changed?

I'm quite sure with this flag during the day the same version of the artifact is returned.
I've noticed today that this isn't the case anymore.

**Scripts used to create container and cause the issue**
_BcContainerHelper version 5.0.1_
```
Get-BCArtifactUrl -country BASE -select Daily
```

Around ±11:00 this morning
```
https://bcartifacts.azureedge.net/sandbox/22.1.55890.56394/base
```

Around ±17:00 this afternoon
```
https://bcartifacts.azureedge.net/sandbox/22.1.55890.56507/base
```

Notice the version x.x.x.56394 is update to 56507. I've expected the same version from 11:00 in the morning as a result.

**Comment by freddydk** [comment]
It hasn't changed - looking at the code this has been the behavior all the time - on the day we ship a new version.
The mechanism is:
- Grab the latest version (now 22.1)
- Grab the latest 22.1 before the date in question.
- if no 22.1 is present - return latest

It would be better to grab first 22.1 after the date in question instead of latest.

**Comment by Arthurvdv** [comment]
Thank you for the feedback! I think I found the unexpected behavior (for me) that is related on the exact day a new version is shipped. 

Our CI/CD pipelines for our BC Cloud customers are configured to run with the -Daily parameter without a specific version. We prep a fresh image every morning and during the day it wil reuse this image for spinning up a docker container. This helps for a shorter processing time on every run.

Also we're using (http://aka.ms/getbc) for every developer to have it own dev environment in Azure and recreating this environment sometimes multiple times a day with these lines below.

```powershell
Import-Module BcContainerHelper -DisableNameChecking
$artifactUrl = Get-BCArtifactUrl -country BASE -select Daily
Replace-BcServerContainer -artifactUrl $artifactUrl
```

In both scenario's it reuses the image during the day, expect on the day a new version of BC is shipped. This is not a huge issue and probably only encounter this twelve times a year.

Is there maybe an other approach to this? Or do you see a viable solution to extend the behavior of the -Daily parameter to return the same build version during the day of the new version on the day of release?

**Comment by freddydk** [comment]
I will consider what we can do

**Comment by freddydk** [comment]
You should use another parameter:

```
Get-BCArtifactUrl -type Sandbox -country us -before ([DateTime]::Today)
```

This will give you the last us artifacts before today - this should always return a value.
You can also do:

```
Get-BCArtifactUrl -type Sandbox -country us -after ([DateTime]::Today) -select first
```

Which will give you the first artifact from today - but that might not always return a value...
For Weekly or monthly use:

```
Get-BCArtifactUrl -type Sandbox -country us -before ([DateTime]::Today.AddDays(-7))
```

or

```
Get-BCArtifactUrl -type Sandbox -country us -before ([DateTime]::Today.AddMonths(-1))
```

**Comment by freddydk** [comment]
Workaround provided and fix shipped in 5.0.2
